blaze/expr/collections.py

```diff
@@ -435,22 +435,31 @@ def schema(self):
         """
         option = lambda dt: dt if isinstance(dt, Option) else Option(dt)
 
-        joined = [[name, dt] for name, dt in self.lhs.schema[0].parameters[0]
-                  if name in self.on_left]
+        on_left = self.on_left
+        on_right = self.on_right
+
+        right_params = self.rhs.schema[0].parameters[0]
```

Comment:
Ok. I'd install spark 1.3 for now, because I haven't worked up a PR to add support for 1.4 yet.